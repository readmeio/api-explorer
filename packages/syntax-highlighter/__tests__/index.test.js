const { mount, shallow } = require('enzyme');
const path = require('path');
const glob = require('glob');
const fs = require('fs').promises;

const syntaxHighlighter = require('..');
const uppercase = require('../uppercase');
const canoncial = require('../canonical');

const fixtures = glob.sync(path.join(__dirname, '/__fixtures__/*'));

test('should highlight a block of code', () => {
  const code = shallow(syntaxHighlighter('var a = 1;', 'javascript'));

  expect(code.hasClass('cm-s-neo')).toBe(true);
  expect(code.html()).toBe(
    '<span class="cm-s-neo"><span class="cm-keyword">var</span> <span class="cm-def">a</span> <span class="cm-operator">=</span> <span class="cm-number">1</span>;</span>'
  );
});

test('should work when passed a non-string value', () => {
  expect(() => syntaxHighlighter(false, 'text')).not.toThrow();
});

test('should sanitize plain text language', () => {
  expect(shallow(syntaxHighlighter('& < > " \' /', 'text')).html()).toContain('&amp; &lt; &gt; &quot; &#x27; /');
});

test('should sanitize mode', () => {
  expect(shallow(syntaxHighlighter('&', 'json')).html()).toContain('&amp;');
  expect(shallow(syntaxHighlighter('<', 'json')).html()).toContain('&lt;');
});

test('should concat the same style items', () => {
  // This is testing the `accum += text;` line
  expect(shallow(syntaxHighlighter('====', 'javascript')).text()).toContain('====');
});

test('should work with modes', () => {
  expect(shallow(syntaxHighlighter('{ "a": 1 }', 'json')).html()).toBe(
    '<span class="cm-s-neo">{ <span class="cm-property">&quot;a&quot;</span>: <span class="cm-number">1</span> }</span>'
  );
});

test('should have a dark theme', () => {
  expect(shallow(syntaxHighlighter('{ "a": 1 }', 'json', { dark: true })).hasClass('cm-s-tomorrow-night')).toBe(true);
});

test('should tokenize variables (double quotes)', () => {
  expect(mount(syntaxHighlighter('"<<apiKey>>"', 'json', { tokenizeVariables: true })).find('Variable')).toHaveLength(
    1
  );
});

test('should tokenize variables (single quotes)', () => {
  expect(mount(syntaxHighlighter("'<<apiKey>>'", 'json', { tokenizeVariables: true })).find('Variable')).toHaveLength(
    1
  );
});

test('should keep enclosing characters around the variable', () => {
  expect(mount(syntaxHighlighter("'<<apiKey>>'", 'json', { tokenizeVariables: true })).text()).toBe("'APIKEY'");
});

describe('Supported languages', () => {
  const languages = fixtures.map(fixture => {
    return [uppercase(path.basename(fixture)), fixture];
  });

  describe.each(languages)('%s', (language, fixtureDir) => {
    let testCase;

    // eslint-disable-next-line global-require, import/no-dynamic-require
    const instructions = require(path.join(fixtureDir, 'index.js'));

    beforeEach(async () => {
      testCase = await fs.readFile(path.join(fixtureDir, `sample.${instructions.mode.primary}`), 'utf8');
    });

    it('should have a properly formatted instruction set', () => {
      expect(instructions).not.toBeUndefined();
      expect(instructions).toStrictEqual({
        language: expect.any(String),
        mode: expect.objectContaining({
          primary: expect.any(String),
          aliases: expect.any(Object),
        }),
      });
    });

    it('should syntax highlight an example', () => {
      const highlighted = shallow(syntaxHighlighter(testCase, instructions.mode.primary)).html();
      expect(highlighted).toMatchSnapshot();
    });

    if (Object.keys(instructions.mode.aliases).length > 0) {
      const aliases = Object.keys(instructions.mode.aliases).map(alias => [alias, instructions.mode.aliases[alias]]);

      describe('Mode aliases', () => {
        describe.each(aliases)('%s', (alias, aliasName) => {
          it('should support the mode alias', () => {
            const highlighted = shallow(syntaxHighlighter(testCase, instructions.mode.primary)).html();
            expect(shallow(syntaxHighlighter(testCase, alias)).html()).toBe(highlighted);
          });

          it('should uppercase the mode alias', () => {
            expect(uppercase(alias)).toBe(aliasName);
          });

          if ('canonical' in instructions.mode) {
            it('should have a canonical directive set up', () => {
              expect(canoncial(alias)).toBe(instructions.mode.canonical);
            });
          } else {
            it('should have a canonical directive set up off the primary mode', () => {
              expect(canoncial(alias)).toBe(instructions.mode.primary);
            });
          }
        });
      });
    }

    if (instructions.mode.primary === 'html') {
      it('should highlight handlebars templates', () => {
        const code = '<p>{{firstname}} {{lastname}}</p>';
        expect(shallow(syntaxHighlighter(code, 'handlebars')).html()).toContain('cm-bracket');
      });
    } else if (instructions.mode.primary === 'php') {
      it('should highlight if missing an opening `<?php` tag', () => {
        expect(shallow(syntaxHighlighter('echo "Hello World";', 'php')).html()).toContain('cm-keyword');
      });
    }
  });
});
