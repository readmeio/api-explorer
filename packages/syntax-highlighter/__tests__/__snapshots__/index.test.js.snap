// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Supported languages ASP.NET should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-operator\\">&lt;%</span><span class=\\"cm-variable\\">@</span> <span class=\\"cm-variable\\">Language</span><span class=\\"cm-operator\\">=</span> <span class=\\"cm-string\\">&quot;VBScript&quot;</span> <span class=\\"cm-operator\\">%&gt;</span>
<span class=\\"cm-operator\\">&lt;%</span>
  <span class=\\"cm-variable\\">Response</span>.<span class=\\"cm-variable\\">Write</span>(<span class=\\"cm-string\\">&quot;Hello World&quot;</span>)
<span class=\\"cm-operator\\">%&gt;</span>
</span>"
`;

exports[`Supported languages C should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-variable\\">#include</span> <span class=\\"cm-operator\\">&lt;</span><span class=\\"cm-variable\\">stdio</span>.<span class=\\"cm-variable\\">h</span><span class=\\"cm-operator\\">&gt;</span>

<span class=\\"cm-variable\\">int</span> <span class=\\"cm-variable\\">main</span>() {
  <span class=\\"cm-variable\\">printf</span>(<span class=\\"cm-string\\">&quot;Hello World\\\\n&quot;</span>);
  <span class=\\"cm-variable\\">return</span> <span class=\\"cm-number\\">0</span>;
}
</span>"
`;

exports[`Supported languages C# should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-keyword\\">class</span> <span class=\\"cm-def\\">HelloWorld</span> {
  <span class=\\"cm-keyword\\">static</span> <span class=\\"cm-keyword\\">void</span> <span class=\\"cm-variable\\">Main</span>() {
    <span class=\\"cm-variable\\">System</span>.<span class=\\"cm-variable\\">Console</span>.<span class=\\"cm-variable\\">WriteLine</span>(<span class=\\"cm-string\\">&quot;Hello World&quot;</span>);
  }
}
</span>"
`;

exports[`Supported languages C++ should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-meta\\">#include &lt;iostream&gt;</span>

<span class=\\"cm-keyword\\">using</span> <span class=\\"cm-keyword\\">namespace</span> <span class=\\"cm-def\\">std</span>;

<span class=\\"cm-type\\">int</span> <span class=\\"cm-def\\">main</span>()
{
  <span class=\\"cm-variable\\">cout</span> <span class=\\"cm-operator\\">&lt;&lt;</span> <span class=\\"cm-string\\">&quot;Hello World\\\\n&quot;</span>;
}
</span>"
`;

exports[`Supported languages CSS should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-tag\\">body</span>::<span class=\\"cm-variable-3\\">before</span> {
  <span class=\\"cm-property\\">content</span>: <span class=\\"cm-string\\">&quot;Hello World&quot;</span>;
}
</span>"
`;

exports[`Supported languages Clojure should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-bracket\\">(</span><span class=\\"cm-keyword\\">println</span> <span class=\\"cm-string\\">&quot;Hello World&quot;</span><span class=\\"cm-bracket\\">)</span>
</span>"
`;

exports[`Supported languages D should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-comment\\">// Hello World in D</span>
<span class=\\"cm-variable\\">import</span> <span class=\\"cm-variable\\">std</span>.<span class=\\"cm-variable\\">stdio</span>;

<span class=\\"cm-variable\\">void</span> <span class=\\"cm-variable\\">main</span>()
{
	   <span class=\\"cm-variable\\">writeln</span>(<span class=\\"cm-string\\">&quot;Hello World&quot;</span>);
}
</span>"
`;

exports[`Supported languages Dart should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-variable\\">main</span>() {
  <span class=\\"cm-variable\\">print</span>(<span class=\\"cm-string\\">&#x27;Hello World&#x27;</span>);
}
</span>"
`;

exports[`Supported languages Diff should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\">diff --git a/index.html b/index.html
index c1d9156..7764744 100644
<span class=\\"cm-negative\\">--- a/index.html</span>
<span class=\\"cm-positive\\">+++ b/index.html</span>
<span class=\\"cm-meta\\">@@ -95,7 +95,8 @@ StringStream.prototype = {</span>
     &lt;script&gt;
       var editor = CodeMirror.fromTextArea(document.getElementById(&quot;code&quot;), {
         lineNumbers: true,
<span class=\\"cm-negative\\">-        autoMatchBrackets: true</span>
<span class=\\"cm-positive\\">+        autoMatchBrackets: true,</span>
<span class=\\"cm-positive\\">+      onGutterClick: function(x){console.log(x);}</span>
       });
     &lt;/script&gt;
   &lt;/body&gt;
</span>"
`;

exports[`Supported languages Dockerfile should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-keyword\\">FROM</span> alpine:3.4
</span>"
`;

exports[`Supported languages Erlang should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-attribute\\">-module</span>(<span class=\\"cm-atom\\">erlang_hw</span>).
<span class=\\"cm-attribute\\">-export</span>([<span class=\\"cm-meta\\">start/0</span>]).

<span class=\\"cm-tag\\">start</span>() -&gt;
  <span class=\\"cm-tag\\">io</span>:<span class=\\"cm-tag\\">format</span>(<span class=\\"cm-string\\">&quot;Hello World~n&quot;</span>).
</span>"
`;

exports[`Supported languages Go should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-keyword\\">package</span> <span class=\\"cm-variable\\">main</span>

<span class=\\"cm-keyword\\">import</span> <span class=\\"cm-string\\">&quot;fmt&quot;</span>

<span class=\\"cm-keyword\\">func</span> <span class=\\"cm-variable\\">main</span>() {
  <span class=\\"cm-variable\\">fmt</span><span class=\\"cm-number\\">.</span><span class=\\"cm-variable\\">Println</span>(<span class=\\"cm-string\\">&quot;Hello World&quot;</span>)
}
</span>"
`;

exports[`Supported languages Groovy should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-variable\\">println</span> <span class=\\"cm-string\\">&quot;Hello World&quot;</span>
</span>"
`;

exports[`Supported languages HTML should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-meta\\">&lt;!DOCTYPE html&gt;</span>
<span class=\\"cm-tag cm-bracket\\">&lt;</span><span class=\\"cm-tag\\">html</span> <span class=\\"cm-attribute\\">lang</span>=<span class=\\"cm-string\\">&quot;en&quot;</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>
<span class=\\"cm-tag cm-bracket\\">&lt;</span><span class=\\"cm-tag\\">head</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>
    <span class=\\"cm-tag cm-bracket\\">&lt;</span><span class=\\"cm-tag\\">meta</span> <span class=\\"cm-attribute\\">charset</span>=<span class=\\"cm-string\\">&quot;UTF-8&quot;</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>
    <span class=\\"cm-tag cm-bracket\\">&lt;</span><span class=\\"cm-tag\\">meta</span> <span class=\\"cm-attribute\\">name</span>=<span class=\\"cm-string\\">&quot;viewport&quot;</span> <span class=\\"cm-attribute\\">content</span>=<span class=\\"cm-string\\">&quot;width=device-width, initial-scale=1.0&quot;</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>
    <span class=\\"cm-tag cm-bracket\\">&lt;</span><span class=\\"cm-tag\\">title</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>Hello World!<span class=\\"cm-tag cm-bracket\\">&lt;/</span><span class=\\"cm-tag\\">title</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>
<span class=\\"cm-tag cm-bracket\\">&lt;/</span><span class=\\"cm-tag\\">head</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>
<span class=\\"cm-tag cm-bracket\\">&lt;</span><span class=\\"cm-tag\\">body</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>
    <span class=\\"cm-tag cm-bracket\\">&lt;</span><span class=\\"cm-tag\\">h1</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>Hello World!<span class=\\"cm-tag cm-bracket\\">&lt;/</span><span class=\\"cm-tag\\">h1</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>
<span class=\\"cm-tag cm-bracket\\">&lt;/</span><span class=\\"cm-tag\\">body</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>
<span class=\\"cm-tag cm-bracket\\">&lt;/</span><span class=\\"cm-tag\\">html</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>
</span>"
`;

exports[`Supported languages HTTP should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-keyword\\">POST</span> <span class=\\"cm-string-2\\">/somewhere</span> <span class=\\"cm-keyword\\">HTTP/1.1</span>
<span class=\\"cm-atom\\">Host:</span><span class=\\"cm-string\\"> example.com</span>
<span class=\\"cm-atom\\">If-Modified-Since:</span><span class=\\"cm-string\\"> Sat, 29 Oct 1994 19:43:31 GMT</span>
<span class=\\"cm-atom\\">Content-Type:</span><span class=\\"cm-string\\"> application/x-www-form-urlencoded;</span>
<span class=\\"cm-string\\">	charset=utf-8</span>
<span class=\\"cm-atom\\">User-Agent:</span><span class=\\"cm-string\\"> Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/536.11 (KHTML, like Gecko) Ubuntu/12.04 Chromium/20.0.1132.47 Chrome/20.0.1132.47 Safari/536.11</span>

This is the request body!
</span>"
`;

exports[`Supported languages JSON should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\">{
  <span class=\\"cm-property\\">&quot;hello&quot;</span>: <span class=\\"cm-string\\">&quot;world&quot;</span>
}
</span>"
`;

exports[`Supported languages Java should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-keyword\\">public</span> <span class=\\"cm-keyword\\">class</span> <span class=\\"cm-def\\">Java</span> {
  <span class=\\"cm-keyword\\">public</span> <span class=\\"cm-keyword\\">static</span> <span class=\\"cm-type\\">void</span> <span class=\\"cm-variable\\">main</span>(<span class=\\"cm-type\\">String</span>[] <span class=\\"cm-variable\\">args</span>) {
    <span class=\\"cm-variable\\">System</span>.<span class=\\"cm-variable\\">out</span>.<span class=\\"cm-variable\\">println</span>(<span class=\\"cm-string\\">&quot;Hello World&quot;</span>);
  }
}
</span>"
`;

exports[`Supported languages JavaScript should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-variable\\">alert</span>(<span class=\\"cm-string\\">&#x27;Hello, world!&#x27;</span>);
</span>"
`;

exports[`Supported languages Julia should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-builtin\\">println</span>(<span class=\\"cm-string\\">&quot;Hello World&quot;</span>)
</span>"
`;

exports[`Supported languages Kotlin should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-keyword\\">package</span> <span class=\\"cm-variable\\">hello</span>

<span class=\\"cm-keyword\\">fun</span> <span class=\\"cm-def\\">main</span>() {
  <span class=\\"cm-variable\\">println</span>(<span class=\\"cm-string\\">&quot;Hello World&quot;</span>)
}
</span>"
`;

exports[`Supported languages Liquid should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\">{% if user %}
  Hello {{ user.name }}!
{% endif %}
</span>"
`;

exports[`Supported languages Markdown should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"># Hello World
</span>"
`;

exports[`Supported languages Objective-C should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-comment\\">/*</span>
 <span class=\\"cm-comment\\">Build on OS X:</span>
 <span class=\\"cm-comment\\">clang -framework Foundation -fobjc-arc objc.m -o objc</span>

 <span class=\\"cm-comment\\">Build on Linux with GNUstep:</span>
 <span class=\\"cm-comment\\">clang \\\\\`gnustep-config --objc-flags\\\\\` \\\\\`gnustep-config --base-libs\\\\\` -fobjc-nonfragile-abi -fobjc-arc objc.m -o objc</span>
 <span class=\\"cm-comment\\">*/</span>

<span class=\\"cm-meta\\">#import &lt;Foundation/Foundation.h&gt;</span>

<span class=\\"cm-type\\">int</span> <span class=\\"cm-def\\">main</span>(<span class=\\"cm-type\\">void</span>)
{
    <span class=\\"cm-variable\\">NSLog</span>(<span class=\\"cm-variable\\">@</span><span class=\\"cm-string\\">&quot;Hello World&quot;</span>);
}
</span>"
`;

exports[`Supported languages Objective-C++ should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-comment\\">/*</span>
 <span class=\\"cm-comment\\">Build on OS X:</span>
 <span class=\\"cm-comment\\">clang -framework Foundation -fobjc-arc objc.m -o objc</span>

 <span class=\\"cm-comment\\">Build on Linux with GNUstep:</span>
 <span class=\\"cm-comment\\">clang \\\\\`gnustep-config --objc-flags\\\\\` \\\\\`gnustep-config --base-libs\\\\\` -fobjc-nonfragile-abi -fobjc-arc objc.m -o objc</span>
 <span class=\\"cm-comment\\">*/</span>

<span class=\\"cm-meta\\">#import &lt;Foundation/Foundation.h&gt;</span>

<span class=\\"cm-type\\">int</span> <span class=\\"cm-def\\">main</span>(<span class=\\"cm-type\\">void</span>)
{
    <span class=\\"cm-variable\\">NSLog</span>(<span class=\\"cm-variable\\">@</span><span class=\\"cm-string\\">&quot;Hello World&quot;</span>);
}
</span>"
`;

exports[`Supported languages PHP should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-operator\\">&lt;?</span><span class=\\"cm-variable\\">php</span>

<span class=\\"cm-keyword\\">echo</span> <span class=\\"cm-string\\">&#x27;Hello World&#x27;</span>;
</span>"
`;

exports[`Supported languages Perl should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-comment\\">#!/usr/bin/perl</span>
<span class=\\"cm-keyword\\">print</span> <span class=\\"cm-string\\">&quot;Hello World\\\\n&quot;</span>;
</span>"
`;

exports[`Supported languages PowerShell should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-string\\">&#x27;Hello World&#x27;</span>
</span>"
`;

exports[`Supported languages Python should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-comment\\">#!/usr/bin/env python</span>
<span class=\\"cm-builtin\\">print</span> <span class=\\"cm-string\\">&quot;Hello World&quot;</span>
</span>"
`;

exports[`Supported languages Ruby should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-comment\\">#!/usr/bin/env ruby</span>
<span class=\\"cm-variable\\">puts</span> <span class=\\"cm-string\\">&quot;Hello World&quot;</span>
</span>"
`;

exports[`Supported languages Rust should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-keyword\\">fn</span> <span class=\\"cm-def\\">main</span>() {
    <span class=\\"cm-variable-3\\">println!</span>(<span class=\\"cm-string\\">&quot;Hello World&quot;</span>);
}
</span>"
`;

exports[`Supported languages SQL should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-keyword\\">SELECT</span> <span class=\\"cm-string\\">&#x27;Hello World&#x27;</span><span class=\\"cm-punctuation\\">;</span>
</span>"
`;

exports[`Supported languages Scala should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-keyword\\">object</span> <span class=\\"cm-def\\">HelloWorld</span> <span class=\\"cm-keyword\\">extends</span> <span class=\\"cm-type\\">App</span> {
  <span class=\\"cm-keyword\\">println</span>(<span class=\\"cm-string\\">&quot;Hello World&quot;</span>)
}
</span>"
`;

exports[`Supported languages Shell should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-meta\\">#!/bin/sh</span>
<span class=\\"cm-builtin\\">echo</span> <span class=\\"cm-string\\">&quot;Hello World&quot;</span>
</span>"
`;

exports[`Supported languages Swift should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-variable\\">print</span><span class=\\"cm-punctuation\\">(</span><span class=\\"cm-string\\">&quot;Hello World&quot;</span><span class=\\"cm-punctuation\\">)</span>
</span>"
`;

exports[`Supported languages TypeScript should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-keyword\\">let</span> { <span class=\\"cm-def\\">a</span>, <span class=\\"cm-def\\">b</span> }: { <span class=\\"cm-property\\">a</span>: <span class=\\"cm-type\\">string</span>, <span class=\\"cm-property\\">b</span>: <span class=\\"cm-type\\">number</span> } <span class=\\"cm-operator\\">=</span> <span class=\\"cm-variable\\">o</span>;
</span>"
`;

exports[`Supported languages XML should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-meta\\">&lt;?xml</span> <span class=\\"cm-meta\\">version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;</span>
<span class=\\"cm-tag cm-bracket\\">&lt;</span><span class=\\"cm-tag\\">text</span><span class=\\"cm-tag cm-bracket\\">&gt;</span><span class=\\"cm-atom\\">&lt;![CDATA[Hello World]]&gt;</span><span class=\\"cm-tag cm-bracket\\">&lt;/</span><span class=\\"cm-tag\\">text</span><span class=\\"cm-tag cm-bracket\\">&gt;</span>
</span>"
`;

exports[`Supported languages YAML should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-atom\\">hello</span><span class=\\"cm-meta\\">: </span>world
</span>"
`;

exports[`Supported languages cURL should syntax highlight an example 1`] = `
"<span class=\\"cm-s-neo\\"><span class=\\"cm-builtin\\">curl</span> <span class=\\"cm-attribute\\">--request</span> GET \\\\
  <span class=\\"cm-attribute\\">--url</span> <span class=\\"cm-string\\">&#x27;https://dash.readme.io/api/v1/api-specification?perPage=10&amp;page=1&#x27;</span> \\\\
  <span class=\\"cm-attribute\\">--header</span> <span class=\\"cm-string\\">&#x27;x-readme-version: v3.0&#x27;</span>
</span>"
`;
